### ISSUE TITLE
Windows 32-bit system will get stuck when executing.

### ISSUE THREAD

**Issue by c17abab** [golang-design/clipboard#45]
The compilation parameters are as follows:
$env:GOARCH="amd64";
$env:GOOS="windows";
$env:GOARCH="386"; # 32
#$env:GOARCH="amd64"; # 64

and the program will be stucked at line 324 in the file clipborad_windows.go: 

	println("test in read 2")
	// try again until open clipboard successed
       // while stuck here
	for {
		r, _, _ = openClipboard.Call()
		if r == 0 {
			continue
		}
		break
	}
        //  the following code will not be executed.
	println("test in read 3")

**Comment by changkun** [comment]
Good point. Is there any specific use case for 32-bit windows? I am 99% sure that this package didn't consider 32-bit windows.

**Comment by c17abab** [comment]
> Good point. Is there any specific use case for 32-bit windows? I am 99% sure that this package didn't consider 32-bit windows.

Found the cause of the problem, the user32.dll function openClipboard need a parameter(A handle to the window to be associated with the open clipboard. If this parameter is NULL, the open clipboard is associated with the current task). So you need get the window handle first：
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-openclipboard

eg:
	// Here you need to get the handle of the clipboard first
	getOpenClipboardWindow := user32.MustFindProc("GetOpenClipboardWindow")
	for {
		r, _, _ = getOpenClipboardWindow.Call()
		r, _, _ = openClipboard.Call(r)
		if r == 0 {
			continue
		}
		break
	}
	defer closeClipboard.Call()

**Comment by changkun** [comment]
Thanks for the investigation. As you already investigated the issue, any PR would be very welcome 👍
